packages/docs/src/components/unobtrusive-consent-control-ui.js

```diff
@@ -0,0 +1,197 @@
+import React, { useCallback, useEffect, useMemo, useState } from 'react'
+import { Form, Field } from 'react-final-form'
+import clsx from 'clsx'
+import { useConsentFormVisible } from '@techboi/consent-manager'
+import { FiChevronUp } from 'react-icons/fi'
+
+import { SwitchAdapter } from './form/switch'
+import styles from './unobtrusive-consent-control-ui.module.css'
+
+export function UnobtrusiveConsentControlUI({
+  integrations,
+  initialValues,
+  onSubmit,
+  introductionNoActionDelay = 2000,
+  introductionTransitionDuration = 1000,
+  introductionVisibleDuration = 4000,
+  ToggleIcon = FiChevronUp,
+}) {
+  const hasPendingDescisions = useConsentFormVisible()
+  const [showIntroduction, setShowIntroduction] = useState(false)
+
+  const [needsIntroduction, setNeedsIntroduction] = useState(
+    hasPendingDescisions
+  )
+  const [slideUp, setSlideUp] = useState(false)
+
+  const toggleControlForm = useCallback(
+    e => {
+      e.preventDefault()
+      setShowIntroduction(false)
+      setSlideUp(v => !v)
+    },
+    [setSlideUp, setShowIntroduction]
+  )
+
+  // @todo calling onSubmit causes rerender
+  const onSubmitCb = useCallback(
+    values => {
+      const enabled = []
+      for (const [key, value] of Object.entries(values)) {
+        if (value) {
+          enabled.push(key)
+        }
+      }
+      setSlideUp(false)
+      onSubmit({ enabled })
+    },
+    [onSubmit]
+  )
+
+  const initial = useMemo(() => {
+    const initialState = {}
+    for (const integration of integrations) {
+      initialState[integration.id] = initialValues.enabled.includes(
+        integration.id
+      )
+    }
+
+    return initialState
+  }, [integrations, initialValues])
+
+  // Check if component was mounted for SSR
+  const [isMounted, setIsMounted] = useState(false)
+  useEffect(() => setIsMounted(true), [setIsMounted])
+
+  // Introduction animation
+  useEffect(() => {
+    if (!isMounted || !needsIntroduction) {
+      return
+    }
+    setShowIntroduction(true)
+    setNeedsIntroduction(false)
+
+    // Wait for no user interaction
+    // @todo actually check it
+    window.setTimeout(() => {
+      setSlideUp(true)
+
+      // Keep it visible to the user
+      window.setTimeout(() => {
+        setSlideUp(false)
+
+        // Wait with swapping contentn till slide down is finished
+        window.setTimeout(() => {
+          setShowIntroduction(false)
```

Comment:
```typescript
const wait = (ms) => new Promise((resolve) => window.setTimeout(resolve, ms))
```

Three levels of callback stuff within an effect looks.. intense..
Wondering whether these need to be sequential in one method or whether we could use some signalling pattern to make it more adaptable.

Reply:
this is quick and dirty prototyping. don't worry, this will be replaced!

I "just" need to find a small alternative to https://greensock.com/docs/v3/GSAP/Tween with a free license